src/core/grid_based_algorithms/lbgpu_cuda.cu

```diff
@@ -2538,6 +2516,11 @@ void lb_init_boundaries_GPU(int host_n_lb_boundaries, int number_of_boundnodes,
   if (this_node != 0)
     return;
 
+  float *boundary_velocity = nullptr;
+  int *boundary_node_list = nullptr;
+  int *boundary_index_list = nullptr;
+  size_t size_of_boundindex = 0;
```

Comment:
These variables shadow globals, and these globals are not even used anymore. It's possible that that was the cause of the `HSA_STATUS_ERROR_INVALID_ISA` you were seeing.

Reply:
[`HSA_STATUS_ERROR_INVALID_ISA`](https://github.com/RadeonOpenCompute/ROCR-Runtime/blob/898beb5adbe227923b84946e9d5f6570e727d34f/src/inc/hsa.h#L192-L195): The instruction set architecture is invalid.

Reply:
That‘s just a generic error code. It is used in many places for which no specific error code has been defined.